test/text-field.html

```diff
@@ -347,6 +347,22 @@
           });
         });
 
+        (IE ? it : it.skip)('should prevent invalid if input event is user originated', done => {
+          textField.preventInvalidInput = true;
+          textField.pattern = '[0-9]*';
+
+          textField.dispatchEvent(new Event('focusin'));
+
+          setTimeout(() => {
```

Comment:
The fix uses raf but the test uses timeout. That sounds like a recipe for a flakey test.

Reply:
Good catch! Replaced with `requestAnimationFrame`.